### ISSUE TITLE
[v2 FEATURE]: allow hpopt to work on multiple splits and ensemble models

### ISSUE THREAD

**Issue by shihchengli** [chemprop/chemprop#981]
**Is your feature request related to a problem? Please describe.**
The `hpopt` CLI can only optimize hyperparameters on a single model. Allowing HPO be run on multiple splits or ensemble models could help to get a better optimized hyperparameters. For example, for model trained small dataset, doing HPO on a single split might be biased. Instead, one can hold out a test set first and use the remaining data to do HPO on multiple replicates. This issue is also related to #974.
